controller/user_blackbox_test.go

```diff
@@ -0,0 +1,156 @@
+package controller_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-wit/app/test"
+	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	"github.com/fabric8-services/fabric8-wit/controller"
+	"github.com/fabric8-services/fabric8-wit/resource"
+	testjwt "github.com/fabric8-services/fabric8-wit/test/jwt"
+	testrecorder "github.com/fabric8-services/fabric8-wit/test/recorder"
+	tf "github.com/fabric8-services/fabric8-wit/test/testfixture"
+	"github.com/goadesign/goa"
+	"github.com/stretchr/testify/suite"
+)
+
+type UserControllerTestSuite struct {
+	gormtestsupport.DBTestSuite
+	config controller.UserControllerConfiguration
+}
+
+type UserControllerConfig struct {
+	url string
+}
+
+func (t UserControllerConfig) GetAuthServiceURL() string {
+	return t.url
+}
+
+func (t UserControllerConfig) GetAuthShortServiceHostName() string {
+	return ""
+}
+
+func (t UserControllerConfig) GetCacheControlUser() string {
+	return ""
+}
+
+func (t UserControllerConfig) IsAuthorizationEnabled() bool {
+	return false
+}
+
+func TestUserController(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &UserControllerTestSuite{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *UserControllerTestSuite) NewSecuredController(options ...configuration.HTTPClientOption) (*goa.Service, *controller.UserController) {
+	svc := goa.New("user-controller")
+	userCtrl := controller.NewUserController(svc, s.GormDB, s.config, options...)
+	return svc, userCtrl
+}
+
+func (s *UserControllerTestSuite) NewUnsecuredController(options ...configuration.HTTPClientOption) (*goa.Service, *controller.UserController) {
+	svc := goa.New("user-controller")
+	userCtrl := controller.NewUserController(svc, s.GormDB, s.config, options...)
+	return svc, userCtrl
+}
+
+func (s *UserControllerTestSuite) TestListSpaces() {
+
+	s.config = UserControllerConfig{
+		url: "https://auth",
+	}
+
+	s.T().Run("ok", func(t *testing.T) {
+
+		t.Run("user has a role in 1 space", func(t *testing.T) {
```

Comment:
Let's have a test for 'user doesn't have a role in any space'?

Reply:
@sbose78 sure, let me add that

Reply:
@sbose78 I've just added the test in 616ae896